runtime-library/CMakeLists.txt

```diff
@@ -0,0 +1,118 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+cmake_minimum_required(VERSION 3.21...3.24)
+
+project(IREE_RUNTIME_LIBRARY)
```

Comment:
We have a few other C/C++/CMake samples in the cpp folder of this repo: https://github.com/iree-org/iree-samples/tree/main/cpp. This looks different enough to warrant its own top level directory, but I do think we are overdue for some general cleanup in this iree-samples repo...

We've been treating this repo as a place to stash code, and with some more purposeful directory structure it could be easier for users to explore, contribute to, and fork for their own projects. See also https://github.com/benvanik/iree-template-runtime-cmake, with the "Use this template" button: 
<img width="341" alt="image" src="https://user-images.githubusercontent.com/4010439/210893984-536c0387-21f1-47ed-b3e4-64f8a17c5012.png">

Maybe this code could be organized into a similar template repository (or folded into the same repository, if generally applicable for enough users)

Reply:
Possibly. But I'd like to discuss with Ben when he's back. We need a story for these specialty integrations without the build system cases and this is as basic as I could make it. Still expect opinions on it.